rmgpy/molecule/group.py

```diff
@@ -1021,6 +1021,33 @@ def update(self):
         self.updateConnectivityValues()
         self.updateFingerprint()
 
+    def updateCharge(self):
+        """
+        Update the charge according to the valence_electron, total bond order, lone pairs
+        and radical electrons. When charged molecule is appearing in reactants, the product
+        template generated by applying recipe will result in wrong charge status. This
+        function is aimed to solve this problem
+        """
+        for atom in self.atoms:
+            if (len(atom.charge) == 1) & (len(atom.lonePairs) == 1) & (len(atom.radicalElectrons) == 1):
+                # if the charge of the group is not labeled, then no charge update will be
+                # performed. If there is a list of charge, no update either. Besides, this groupatom have to
```

Comment:
'If there is a list of charge, no update either.' - Do you mean a list of more than one entry?

Reply:
Yes... sorry for unclear word.